src/main/java/com/google/devtools/build/lib/remote/downloader/GrpcRemoteDownloader.java

```diff
@@ -235,6 +238,11 @@ static FetchBlobRequest newFetchBlobRequest(
               .setName(QUALIFIER_CHECKSUM_SRI)
               .setValue(checksum.get().toSubresourceIntegrity())
               .build());
+    } else {
+      // If no checksum is provided, never accept cached content.
+      // Timestamp is offset by an hour to account for clock skew.
+      long milis = BlazeClock.instance().currentTimeMillis() + Duration.ofHours(1).toMillis();
+      requestBuilder.setOldestContentAccepted(Timestamps.fromMillis(milis));
```

Comment:
Doing the arithmetic on `Instant` would be slightly cleaner:
```suggestion
      requestBuilder.setOldestContentAccepted(
          Timestamps.fromMillis(
              BlazeClock.instance().now().plus(Duration.ofHours(1)).toEpochMilli()));
```